docs/development/guidelines.md

```diff
@@ -370,7 +429,43 @@ Examples
 
 ## Tests
 
-If a function contains more code than just the getting or setting of a value, automated test should be added to the [`tests`][tests-folder] folder. The file structure in the tests folder should mirror the file structure of the [`src`][src-folder] folder.
+If a function contains more code
+than just the getting or setting of a value,
+automated test should be added
+to the [`tests`][tests-folder] folder.
+The file structure in the tests folder
+should mirror the file structure
+of the [`src`][src-folder] folder.
+
+Names of test functions
+shall start with `test_should_`
+followed by a description
+of the expected behaviour,
+e.g. `test_should_add_column`.
+
+### Testcases
+
+Tests should be parametrized
+using `@pytest.mark.parametrize`,
+even if there is only a single testcase.
+Testcases should be given
+descriptive IDs.
+
+Example:
+
+    ```py
+    @pytest.mark.parametrize(
+        ("path", "expected"),
+        [
+            ("table.csv", Table({"A": [1], "B": [2]})),
+            (Path("table.csv"), Table({"A": [1], "B": [2]})),
+            ("empty_table.csv", Table()),
+        ],
+        ids=["by string", "by path", "empty"],
+    )
+    def test_should_create_table_from_csv_file(path: str | Path, expected: Table) -> None:
+        ...
+    ```
```

Comment:
Maybe have an example specifically for the parametrization and using the do/don't schema from above?